feat(client): ✨ Add Splash into Dev

gchamb opened the pull request:
Merging the splash branch into dev so everyone can use it.

gchamb committed:
feat(client): ✨ Create Splash Page

Started on splash page, Firestore functions for updating display name,  and started on menu.

#26

gchamb committed:
feat: Add Firebase Rule for Users

- Created firebase rule that allows the user itself to write to their document
but allow anyone to read from it (Maybe Change lately).

gchamb committed:
feat(client): :sparkles: Add Modal Component

- Created Modal Component
- Used Modal Component for Rules Modal

gchamb committed:
Merge branch 'development' into 26-splash-screen

gchamb committed:
refactor: Use the firestore types

- rename splashFunctions to just splash
- use the firestore types
- relocated the Modal to appropriate component folder

SBAsai committed:
feat(client): ✨Add auth checking

SBAsai committed:
fix: client test

-Fixed the playwright test expecting the default page

SBAsai committed:
fix(client): :ambulance: Fix account dropdown

-Fixed account dropdown to now work properly

gchamb committed:
Merge branch '26-splash-screen' of https://github.com/cat-acronym-team/cat-me-if-you-can into 26-splash-screen

gchamb committed:
fix: Question Mark for layout

- question mark on each page
- removed styles from html
- cleaned up code changing display name

gchamb committed:
fix: Linting error for empty function

gchamb committed:
Merge branch 'development' into 26-splash-screen

gchamb committed:
feat(client): ✨ Create Anon User

- When the create lobby is created then create anon user and set its display name
- Fixed Modal

nstringham commented on client/src/routes/+page.svelte:
```diff
@@ -1,2 +1,231 @@
+  import { getAuth } from "firebase/auth";
+  import { goto } from "$app/navigation";
+
+  const auth = getAuth();
```
this should be imported from auth.ts

nstringham commented on client/src/routes/+page.svelte:
```diff
@@ -1,2 +1,231 @@
+      saveDisplayName(user.uid, name);
+    }
+  };
+  const createLobbyHandler = async () => {
+    // if user is null create anon user
+    // then create a document with the id of the anon user if it doesn't exist create one
+    if (user === null && name !== "") {
+      const { uid } = (await loginAnonymous()).user;
+      createAnonUser(uid, name);
+    }
+
+    if (name !== "") {
+      // TODO: Create Function Here
+
+      // TODO: Push to game page with code
+      goto("/");
+    }
+  };
```
maybe do an early return if name is "" instead of checkin for it twice

nstringham commented on client/src/lib/firebase/splash.ts:
```diff
@@ -0,0 +1,27 @@
+export const createAnonUser = async (uid: string, name: string) => {
+  await setDoc(doc(db, `/users/${uid}`), {
+    displayName: name,
+    avatar: 1,
```
this should actually be a 0 sorry I said 1 earlier

nstringham commented on client/src/lib/firebase/splash.ts:
```diff
@@ -0,0 +1,27 @@
+};
+
+// create user document for anon user
+export const createAnonUser = async (uid: string, name: string) => {
```
why is this called createAnonUser? isn't is setting a doc for any user?

gchamb replied:
I assumed that if they signed in they should already have a user document in the users collection but your suggestion is good since it wasn't stated

nstringham commented on firestore.rules:
```diff
@@ -6,4 +6,10 @@ service cloud.firestore {
           request.time < timestamp.date(2022, 10, 20);
     }
   }
+  match /users/{userId} {
+        // only allows the itself to write to the document
+        allow write: if request.auth.uid == userId;
+        // allows anyone to read that information
+        allow read;
+  }
```
this should be inside /databases/{database}/documents

gchamb committed:
fix(client): 🐛 Implement review requests

- Add firebase rule(regex needs finishing)
- Changed create user for any user not just anon
